src/DataStructures/Variables.hpp

```diff
@@ -152,9 +170,9 @@ class Variables<tmpl::list<Tags...>> {
 
   /// Initialize a Variables to the state it would have after calling
   /// the constructor with the same arguments.
-  void initialize(
-      size_t number_of_grid_points,
-      double value = std::numeric_limits<double>::signaling_NaN()) noexcept;
+  void initialize(size_t number_of_grid_points, double value) noexcept;
+
+  void initialize(size_t number_of_grid_points) noexcept;
```

Comment:
Should these two be in a doxygen group?

Also everywhere else you put the single-argument function before the two-argument function, so consider reordering for consistency.